kolibri/core/discovery/models.py

```diff
@@ -37,4 +49,48 @@ def available(self):
             self.save()
             return True
         except NetworkClientError:
+            if self.dynamic:
+                self.delete()
             return False
+
+
+class StaticNetworkLocationManager(models.Manager):
+    def get_queryset(self):
+        queryset = super(StaticNetworkLocationManager, self).get_queryset()
+        return queryset.filter(dynamic=False)
+
+
+class StaticNetworkLocation(NetworkLocation):
+    objects = StaticNetworkLocationManager()
+
+    class Meta:
+        proxy = True
+
+    def save(self, *args, **kwargs):
+        self.dynamic = False
+        return super(StaticNetworkLocation, self).save(*args, **kwargs)
+
+
+class DynamicNetworkLocationManager(models.Manager):
+    def get_queryset(self):
+        queryset = super(DynamicNetworkLocationManager, self).get_queryset()
+        return queryset.filter(dynamic=True)
```

Comment:
at some point, this could be extended to dynamically filter out `DynamicNetworkLocation` objects with a redundant `StaticNetworkLocation` in existence (i.e. one with the same `instance_id`)

Reply:
this is no longer relevant